### PR TITLE
Implement MBAAS Auth iOS

### PR DESCRIPTION
**Pull request by jackycodes**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

[AB#62459](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/62459)

Implemented the following on iOS 
`isEnabled`
`setEnabled`
`signIn`
`signOut`

## Related PRs or issues

Note: This PR depends on [594](https://github.com/microsoft/appcenter-sdk-react-native/pull/594)

### CODE REVIEW

File: appcenter-auth/ios/AppCenterReactNativeAuth/AppCenterReactNativeAuth.m [hunk 1]
```diff
@@ -40,4 +43,35 @@ + (void)register {
...
+    [MSAuth setEnabled:shouldEnable];
+    resolve(nil);
+}
+
+RCT_EXPORT_METHOD(signIn:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject) {
+    [MSAuth signInWithCompletionHandler:^(MSUserInformation * _Nullable userInformation, NSError * _Nullable error) {
+        if(!error) {
+            
+            /* Sign-in succeeded, convert Java result to a JavaScript result. */
```
Comment (guperrot) [inline appcenter-auth/ios/AppCenterReactNativeAuth/AppCenterReactNativeAuth.m]:
```suggestion
            /* Sign-in succeeded, convert native result to a JavaScript result. */
```

File: appcenter-auth/ios/AppCenterReactNativeAuth/AppCenterReactNativeAuth.m [hunk 2]
```diff
@@ -40,4 +43,35 @@ + (void)register {
...
+                  resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject) {
+    [MSAuth setEnabled:shouldEnable];
+    resolve(nil);
+}
+
+RCT_EXPORT_METHOD(signIn:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject) {
+    [MSAuth signInWithCompletionHandler:^(MSUserInformation * _Nullable userInformation, NSError * _Nullable error) {
+        if(!error) {
```
Comment (dhei) [inline appcenter-auth/ios/AppCenterReactNativeAuth/AppCenterReactNativeAuth.m]:
```suggestion
        if (!error) {
```

File: appcenter-auth/ios/AppCenterReactNativeAuth/AppCenterReactNativeAuth.m [hunk 3]
```diff
@@ -40,4 +43,35 @@ + (void)register {
...
+            
+            /* Sign-in succeeded, convert Java result to a JavaScript result. */
+            NSMutableDictionary *dict = [[NSMutableDictionary alloc] init];
+            dict[kMSAccountId] = userInformation.accountId;
```
Comment (dhei) [inline appcenter-auth/ios/AppCenterReactNativeAuth/AppCenterReactNativeAuth.m]: What about id token and access token?

Comment (guperrot) [reply appcenter-auth/ios/AppCenterReactNativeAuth/AppCenterReactNativeAuth.m]: It needs the 2.1.0 SDK or use a prerelease. Not sure if we can easily commit that (android was easy to consume prerelease).

**Comment by jackycodes** [comment]
> Should you target android PR then? For the base in github.

I was hoping to merge the Android branch into mbaas first.

File: appcenter-auth/ios/AppCenterReactNativeAuth/AppCenterReactNativeAuth.m (hunk 1 above)
Comment (jackycodes) [reply appcenter-auth/ios/AppCenterReactNativeAuth/AppCenterReactNativeAuth.m]: Fixed

File: appcenter-auth/ios/AppCenterReactNativeAuth/AppCenterReactNativeAuth.m (hunk 2 above)
Comment (jackycodes) [reply appcenter-auth/ios/AppCenterReactNativeAuth/AppCenterReactNativeAuth.m]: Fixed

File: appcenter-auth/ios/AppCenterReactNativeAuth/AppCenterReactNativeAuth.m (hunk 3 above)
Comment (dhei) [reply appcenter-auth/ios/AppCenterReactNativeAuth/AppCenterReactNativeAuth.m]: Using prerelease iOS sdk is doable but requires some extra steps compared to android. We should do it.

**Comment by dhei** [comment]
Can remove the "do not merge" label after integrated prerelease native iOS sdk.

**Pull request merged**